dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift

```diff
@@ -8,39 +8,93 @@
 
 import UIKit
 
-class CreatePostViewController: UIViewController {
-	
-	@IBOutlet weak var textView: UITextView!
-	@IBOutlet weak var doneButton: UIButton!
-	
-	override func viewDidLoad() {
-		super.viewDidLoad()
-		
-		addTargets()
-	}
-	
-	private func addTargets() {
-		doneButton.addTarget(self, action: #selector(doneButtonClicked), for: .touchUpInside)
-	}
-	
-	@objc private func doneButtonClicked() {
-		guard let text = textView.text, !text.isEmpty else {
-			showAlert(with: "Error", and: "Text is empty")
-			return
-		}
-		
-		guard let user = AuthManager.shared.currentUser else {
-			showAlert(with: "Error", and: "User not logged in")
-			return
-		}
-		
-		PostManager.shared.createPost(from: user, with: text) { (result) in
-			switch result {
-			case .error(let textError):
-				self.showAlert(with: "Errror", and: textError)
-			case .success:
-				self.showAlert(with: "Success", and: "Post has been created")
-			}
-		}
-	}
+final class CreatePostViewController: UIViewController {
+
+    @IBOutlet private weak var textView: UITextView!
+    @IBOutlet private weak var doneButton: UIButton!
+    @IBOutlet private weak var postImageView: UIImageView!
+    @IBOutlet private weak var crossImageView: UIImageView!
+
+    private let tapToAddPostImage = #imageLiteral(resourceName: "tap_button")
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        Decorator.decorate(vc: self)
+
+        addTargets()
+        addGestures()
+
+        updatePostImageView(image: tapToAddPostImage)
+    }
+}
+
+private extension CreatePostViewController {
+
+    func addTargets() {
+        doneButton.addTarget(self, action: #selector(doneButtonClicked), for: .touchUpInside)
+    }
+
+    func addGestures() {
+        let postImageTapGesture = UITapGestureRecognizer(target: self, action: #selector(postImageClicked))
+        postImageView.addGestureRecognizer(postImageTapGesture)
+
+        let crossTapGesture = UITapGestureRecognizer(target: self, action: #selector(crossImageClicked))
+        crossImageView.addGestureRecognizer(crossTapGesture)
+    }
+
+    @objc func postImageClicked() {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.delegate = self
+        imagePickerController.sourceType = .photoLibrary
+        present(imagePickerController, animated: true, completion: nil)
```

Comment:
needs to update `present` for iPad

Reply:
I don't remember. Maybe it needs `popoverPresentationController.sourceView` like for the alerts.